assets/components/MainMenu.svelte

```diff
@@ -152,9 +154,21 @@ function onSearchKeydown(e) {
 function renderUnread(conversation, max = 60) {
   return conversation.unread > max ? (max + '+') : conversation.unread || 0;
 }
+
+function toggleSection(connection_id) {
+    if (collapsedStates[connection_id]) {
+        collapsedStates[connection_id] = !collapsedStates[connection_id];
+    } else {
+        collapsedStates[connection_id] = true; // default to collapsed if not already in the object
+    }
```

Comment:
Don't think you need all the if/else logic. Just do:

```javascript
function toggleSection(connection_id) {
  collapsedStates[connection_id] = !collapsedStates[connection_id];
  localStorage.setItem('collapsedStates', JSON.stringify(collapsedStates));
}
```

Or do you want the sections to be collapsed as default..? I'm not very fond of hiding information from new users.

Reply:
I think you're right. If there is no cookie set, then all sections should be expanded, else collapse the ones that need to be collapsed. I've updated the code with your suggestion.